sc-kpm/sc-python/services/http_api/auth/auth_handlers.py

```diff
@@ -0,0 +1,135 @@
+                },
+                cnt.REFRESH_TOKEN: {
+                    cnt.TOKEN: refresh_token_data.decode(),
+                    cnt.TOKEN_TYPE: cnt.JWT,
+                    cnt.EXPIRES_IN: params[cnt.REFRESH_TOKEN_LIFE_SPAN],
+                }
+            })
+        else:
+            response = get_response_message(params[cnt.MSG_CODES][cnt.MSG_USER_NOT_FOUND])
+        self.write(response)
+
+    def _get_user_credentials(self):
+        data = json.loads(self.request.body)
+        username = data[cnt.USERNAME] if cnt.USERNAME in data else False
+        password = data[cnt.PASSWORD] if cnt.PASSWORD in data else False
+        return username, password
+
+
+class GetAccessTokenHandler(BaseHandler):
+    @TokenValidator.validate_typed_token(TokenType.REFRESH)
+    def post(self) -> None:
+        access_token_data = _generate_token(TokenType.ACCESS)
+        response = json.dumps({
+            cnt.MSG_CODE: params[cnt.MSG_CODES][cnt.MSG_ALL_DONE],
+            cnt.TOKEN: access_token_data.decode(),
+            cnt.TOKEN_TYPE: cnt.JWT,
+            cnt.EXPIRES_IN: params[cnt.ACCESS_TOKEN_LIFE_SPAN],
+        })
+        self.write(response)
+
+
+class AddUserHandler(BaseHandler):
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def post(self) -> None:
+        database = DataBase()
+        request_params = self._get_request_params([cnt.USERNAME, cnt.PASSWORD, cnt.ROLE])
+        if not username_verifier.validate(request_params[cnt.USERNAME]):
+            response = get_response_message(params[cnt.MSG_CODES][cnt.MSG_INVALID_USERNAME])
+        elif database.is_such_user_in_base(request_params[cnt.USERNAME]):
+            response = get_response_message(params[cnt.MSG_CODES][cnt.MSG_USER_IS_IN_BASE])
+        else:
+            user_added = database.add_user(**request_params)
+            all_done_mes = params[cnt.MSG_CODES][cnt.MSG_ALL_DONE]
+            invalid_role_mes = params[cnt.MSG_CODES][cnt.MSG_INVALID_ROLE]
+            response = get_response_message(all_done_mes) if user_added else get_response_message(invalid_role_mes)
+        self.write(response)
+
+
+class DeleteUserHandler(BaseHandler):
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def post(self) -> None:
+        database = DataBase()
+        request_params = self._get_request_params([cnt.USER_ID])
+        database.delete_user_by_id(request_params[cnt.USER_ID])
+        response = get_response_message(params[cnt.MSG_CODES][cnt.MSG_ALL_DONE])
+        self.write(response)
+
+
+class UpdateUserHandler(BaseHandler):
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def post(self) -> None:
+        database = DataBase()
+        post_args = self._get_request_params([cnt.USER_ID, cnt.USERNAME, cnt.PASSWORD, cnt.ROLE])
+        database.update_user_by_id(**post_args)
+        response = get_response_message(params[cnt.MSG_CODES][cnt.MSG_ALL_DONE])
+        self.write(response)
```

Comment:
I think you can define UserHandler and implement "post" method for creating user, "delete" for deleting and for update you can [choose between "patch" and "put" methods](https://www.baeldung.com/http-put-patch-difference-spring).

Reply:
Same thing with endpoint, you can define single endpoint for user without specifying actions in URL.